Engine/Cli/RunCliAction.cs

```diff
@@ -148,9 +148,13 @@ public int Execute(CancellationToken cancellationToken)
             }
 
 
-            if (!NonInteractive && UserInput.Interface == null)
+            if (UserInput.Interface == null)
```

Comment:
I like the idea of the 4 modes in your description. Why don't we make those 4 modes arguments for CliUserInputInterface to make it explicit when the class is created and move that state more up?